src/field_theory/intermediate_field.lean

```diff
@@ -327,6 +327,27 @@ begin
     rw [← aeval_def, aeval_coe, aeval_def, hProot, add_submonoid_class.coe_zero] },
 end
 
+/-- The map `E → F` when `E` is an intermediate field contained in the intermediate field `F`.
+
+This is the intermediate field version of `subalgebra.inclusion`. -/
+def inclusion {E F : intermediate_field K L} (hEF : E ≤ F) : E →ₐ[K] F :=
+subalgebra.inclusion hEF
+
+lemma inclusion_injective {E F : intermediate_field K L} (hEF : E ≤ F) :
+  function.injective (inclusion hEF) :=
+λ _ _, subtype.ext ∘ subtype.mk.inj
+
+@[simp] lemma inclusion_self {E : intermediate_field K L}:
+  inclusion (le_refl E) = alg_hom.id K E :=
+alg_hom.ext $ λ x, subtype.ext rfl
+
+@[simp] lemma inclusion_inclusion {E F G : intermediate_field K L} (hEF : E ≤ F) (hFG : F ≤ G)
+  (x : E) : inclusion hFG (inclusion hEF x) = inclusion (le_trans hEF hFG) x :=
+subtype.ext rfl
```

Comment:
All of these should have a one-term proof in terms of the `subalgebra` version; which is worth usning not only because it is shorter, but because it ensures that the API of the two versions stays in sync.
```suggestion
lemma inclusion_injective {E F : intermediate_field K L} (hEF : E ≤ F) :
  function.injective (inclusion hEF) :=
subalgebra.inclusion_injective hEF

@[simp] lemma inclusion_self {E : intermediate_field K L}:
  inclusion (le_refl E) = alg_hom.id K E :=
subalgebra.inclusion_self

@[simp] lemma inclusion_inclusion {E F G : intermediate_field K L} (hEF : E ≤ F) (hFG : F ≤ G)
  (x : E) : inclusion hFG (inclusion hEF x) = inclusion (le_trans hEF hFG) x :=
subalgebra.inclusion_inclusion hEF hFG
```
(not tested)

Reply:
You got 2/3 ;-) Need an extra `x` on line 346. I've changed the proofs to these.